plugin/pkg/shoot/validator/admission.go

```diff
@@ -660,11 +660,9 @@ func (c *validationContext) validateShootNetworks(workerless bool) field.ErrorLi
 	)
 
 	if c.seed != nil {
-		// if the shoot is workerless and doesn't have networking field set yet, but the seed has
-		// shootDefaults set for serviceCIDR, then default networking field
-		if workerless &&
-			c.seed.Spec.Networks.ShootDefaults != nil &&
-			c.seed.Spec.Networks.ShootDefaults.Services != nil {
+		// if the shoot is workerless and doesn't have networking field set yet, then
+		// set shoot IPFamilies of seed IPFamilies type.
```

Comment:
```suggestion
		// set shoot IPFamilies to seed IPFamilies type.
```

Reply:
Since we don't allow Shoots without serviceCIDRs or default them from seed always, then can't we default networking field in static defaulting itself?

Reply:
Why don't we default them from the seed? We still need to ensure disjointedness of the CIDRs of seed and shoot, no?

Reply:
We follow whatever defaulting is present for normal shoots. Nothing special if workerless.

Reply:
OK, it seems the commit message of https://github.com/gardener/gardener/pull/7828/commits/a071a4dd08fc0e6bca7d473104a3cd30d303ea5a caused confusion. It's only about the top level `Networking` struct, not about the CIDRs themselves. Then it should be fine